### PR TITLE
port over rest of the jobs/config_test.py logic

### PR DESCRIPTION
**Pull request by krzyzacy**
also dropped some deprecated flag checks.. also we won't need config_sort.. 

/area jobs
/assign @BenTheElder

### CODE REVIEW

**Comment by BenTheElder** [comment]
> /lint

doing my job for me? 😂

**Comment by krzyzacy** [comment]
you mean @fejta-bot's job?

File: config/tests/jobs/jobs_test.go [hunk 1]
```diff
@@ -774,7 +774,89 @@ func checkScenarioArgs(jobName string, args []string) error {
...
 	if use_shared_build_in_args && build_in_args {
-		return fmt.Errorf("Job %s: --use-shared-build and --build cannot be combined", jobName)
+		return fmt.Errorf("job %s: --use-shared-build and --build cannot be combined", jobName)
+	}
+
+	if scenario != "kubernetes_e2e" {
+		return nil
+	}
+
+	if hasArg("--provider=gke", args) {
```
Comment (BenTheElder) [inline config/tests/jobs/jobs_test.go]: it probably doesn't matter given how small the typical list of args is but I feel bad about how many times we loop over the args to check if an arg exists... I instinctively want to make a map and check that

File: config/tests/jobs/jobs_test.go [hunk 2]
```diff
@@ -774,7 +774,89 @@ func checkScenarioArgs(jobName string, args []string) error {
...
+	}
+
+	expectedExtract := 1
+	if sharedBuilds || nodeE2e || builds {
+		expectedExtract = 0
+	} else if strings.Contains(jobName, "upgrade") ||
+		strings.Contains(jobName, "skew") ||
+		strings.Contains(jobName, "rollback") ||
+		strings.Contains(jobName, "downgrade") ||
+		jobName == "ci-kubernetes-e2e-gce-canary" {
```
Comment (BenTheElder) [inline config/tests/jobs/jobs_test.go]: `|| jobName == "ci-kubernetes-e2e-gce-canary` .... ok, I don't blame this at all, but this is sorta gross 😬

File: config/tests/jobs/jobs_test.go [hunk 3]
```diff
@@ -774,7 +774,89 @@ func checkScenarioArgs(jobName string, args []string) error {
...
+	}
+
+	numExtract := 0
+	for _, arg := range args {
+		if strings.HasPrefix(arg, "--extract=") {
+			numExtract++
+		}
+	}
+	if numExtract != expectedExtract {
+		return fmt.Errorf("e2e jobs %s should have %d --extract flags, got %d", jobName, expectedExtract, numExtract)
```
Comment (BenTheElder) [inline config/tests/jobs/jobs_test.go]: hmm this is going to be a confusing error message given that it say you should have n flags for a job type but it varies...

File: config/tests/jobs/jobs_test.go (hunk 2 above)
Comment (krzyzacy) [reply config/tests/jobs/jobs_test.go]: /shrug
yeah I did a 1:1 port for now...

**Comment by BenTheElder** [comment]
> you mean @fejta-bot's job?

@fejta-bot doesn't `/lint` ... yet 🤔

**Comment by krzyzacy** [comment]
I feel like this doesn't worth refactoring, and we can probably get rid of most of them with some kubetest flag redesign...

**Comment by krzyzacy** [comment]
I'll probably rewrite this for a bit after the migration is done
/hold cancel

File: config/jobs/kubernetes/sig-testing/kubetest-canaries.yaml [hunk 4]
```diff
@@ -296,7 +296,6 @@ periodics:
       - --timeout=260
       - --scenario=kubernetes_e2e
       - --
-      - --build=bazel
```
Comment (krzyzacy) [inline config/jobs/kubernetes/sig-testing/kubetest-canaries.yaml]: this is actually invalid... we should handle build within kubemark logic

**Comment by krzyzacy** [comment]
fixed some invalid config
/shrug

**Pull request merged**